lib/grasshopper/grasshopper.c

```diff
@@ -273,52 +288,52 @@ static unsigned char poly_mul(unsigned char left, unsigned char right) {
   return ret;
 }
 
-/* Pregenerations.  */
-static __attribute__((noinline)) void generate_mul_tbl() {
-  for (int i = 0; i < 256; ++i)
-    for (int j = 0; j < 16; ++j)
-      Muls[i][j] = poly_mul(i, Lin[j]);
-}
-
-static __attribute__((noinline)) void generate_coef_tbl() {
-  for (int i = 0; i < NUM_ROUNDS / 2 - 1; ++i)
-    for (int j = 0; j < BIT_IN_BYTES; ++j) {
-      Block *c = &Coefs[i][j];
-      memset(c, 0, BLOCK_SIZE);
-      c->data[BLOCK_SIZE - 1] = i * 8 + j + 1;
-      apply_l(c);
-    }
-}
-
-static __attribute__((noinline)) void generate_LS_tbl() {
-#define MATR_POWER 4
-  unsigned char l_tbl[BLOCK_SIZE][BLOCK_SIZE] = {0};
-
-  /* Init matrix.  */
-  for (int i = 0; i < BLOCK_SIZE; ++i)
-    for (int j = 0; j < BLOCK_SIZE; ++j)
-      if (i - 1 == j)
-        l_tbl[i][j] = 1;
-      else if (i == 0)
-        l_tbl[i][j] = Lin[j];
-
-  /* Powering the matrix.  */
-  unsigned char tmp[BLOCK_SIZE][BLOCK_SIZE] = {0};
-  for (int p = 0; p < MATR_POWER; ++p) {
-    for (int i = 0; i < BLOCK_SIZE; ++i)
-      for (int j = 0; j < BLOCK_SIZE; ++j)
-        for (int k = 0; k < BLOCK_SIZE; ++k)
-          tmp[i][j] ^= poly_mul(l_tbl[i][k], l_tbl[k][j]);
-
-    memcpy(l_tbl, tmp, 256 * sizeof(unsigned char));
-    memset(tmp, 0, 256 * sizeof(unsigned char));
-  }
-
-  for (int i = 0; i < BLOCK_SIZE; ++i)
-    for (int j = 0; j < 256; ++j)
-      for (int k = 0; k < BLOCK_SIZE; ++k)
-        LS_tbl[i][j][k] ^= poly_mul(S[j], l_tbl[k][i]);
-}
+// /* Pregenerations.  */
```

Comment:
Remove this functions (which are comments now).

Moreover, in the future, if you'd like to switch the code off it's better to use preprocessor directives. e.g.:

```
/*
DEAD CODE COULD BE HERE BUT THIS IS NOT THE BEST WAY
*/
#if 0
DEAD CODE COULD BE HERE AND THIS IS THE BEST WAY
#endif
```